staging/src/k8s.io/kubectl/pkg/drain/drain.go

```diff
@@ -316,6 +316,9 @@ func (d *Helper) evictPods(pods []corev1.Pod, evictionGroupVersion schema.GroupV
 					returnCh <- nil
 					return
 				} else if apierrors.IsTooManyRequests(err) {
+					if d.OnPodDeletionOrEvictionFinished != nil {
```

Comment:
This will be retried as well and PDB may not be the only reason and it may heal after a couple of retries. But I understand the inconvenience in here and in case of the PDB blockages, it is better to invoke the finished hook.